File: server/src/routes/inquiry.ts
import { Router } from 'express';
import pool from '../config/db';
import { requireAuth, AuthRequest } from '../middleware/auth';

const router = Router();

// POST /api/inquiries — submit inquiry
router.post('/', requireAuth, async (req: AuthRequest, res) => {
  try {
    const { title, content } = req.body;
    if (!title || !content) {
      res.status(400).json({ error: 'Title and content are required' });
      return;
    }

    const result = await pool.query(
      `INSERT INTO inquiries (user_id, title, content)
       VALUES ($1, $2, $3)
       RETURNING *`,
      [req.uid, title, content],
    );

    res.status(201).json(result.rows[0]);
  } catch (err) {
    console.error('Error creating inquiry:', err);
    res.status(500).json({ error: 'Failed to create inquiry' });
  }
});

// GET /api/inquiries/my — my inquiries
router.get('/my', requireAuth, async (req: AuthRequest, res) => {
  try {
    const result = await pool.query(
      `SELECT * FROM inquiries WHERE user_id = $1 ORDER BY created_at DESC`,
      [req.uid],
    );

    res.json(result.rows);
  } catch (err) {
    console.error('Error fetching inquiries:', err);
    res.status(500).json({ error: 'Failed to fetch inquiries' });
  }
});

export default router;

File: mobile/lib/models/inquiry.dart
class Inquiry {
  final int id;
  final String userId;
  final String title;
  final String content;
  final String status;
  final String? adminReply;
  final DateTime? repliedAt;
  final DateTime createdAt;
  final String? email;
  final String? displayName;

  Inquiry({
    required this.id,
    required this.userId,
    required this.title,
    required this.content,
    required this.status,
    this.adminReply,
    this.repliedAt,
    required this.createdAt,
    this.email,
    this.displayName,
  });

  factory Inquiry.fromJson(Map<String, dynamic> json) {
    return Inquiry(
      id: json['id'],
      userId: json['user_id'],
      title: json['title'],
      content: json['content'],
      status: json['status'] ?? 'pending',
      adminReply: json['admin_reply'],
      repliedAt: json['replied_at'] != null
          ? DateTime.parse(json['replied_at'])
          : null,
      createdAt: DateTime.parse(json['created_at']),
      email: json['email'],
      displayName: json['display_name'],
    );
  }

  String get statusLabel {
    switch (status) {
      case 'replied':
        return '답변완료';
      case 'pending':
      default:
        return '대기중';
    }
  }
}
